Maple2.File.Parser/ServerTableParser.cs

```diff
@@ -367,4 +377,65 @@ public class ServerTableParser {
             yield return (individualItemDrop.dropGroupID, individualItemDrop);
         }
     }
+
+    public IEnumerable<(int Id, RandomRoom RandmRoom)> ParseRoomRandom() {
+        XmlReader reader = xmlReader.GetXmlReader(xmlReader.GetEntry("table/Server/room_random.xml"));
+        var data = roomRandomSerializer.Deserialize(reader) as RoomRandomRoot;
+        Debug.Assert(data != null);
+
+        foreach (RandomRoom roomRandom in data.randomroom) {
+            yield return (roomRandom.id, roomRandom);
+        }
+    }
+
+    public IEnumerable<(int Id, RoomRandom Room)> ParseRoom() {
+        XmlReader reader = xmlReader.GetXmlReader(xmlReader.GetEntry("table/Server/room_random.xml"));
+        var data = roomRandomSerializer.Deserialize(reader) as RoomRandomRoot;
+        Debug.Assert(data != null);
+
+        foreach (RoomRandom roomRandom in data.room) {
+            yield return (roomRandom.id, roomRandom);
+        }
+    }
```

Comment:
parsing the same xml twice?

Reply:
yeah. The xml has two sections. One for room random probability and one for room info